### PR TITLE
PyPy fixes

### PR DESCRIPTION
**Pull request by isuruf**
See commit messages for details.

### CODE REVIEW

File: conda_build/metadata.py [hunk 1]
```diff
@@ -105,7 +105,11 @@ def ns_cfg(config):
                     py33=bool(py == 33),
                     py34=bool(py == 34),
                     py35=bool(py == 35),
-                    py36=bool(py == 36),))
+                    py36=bool(py == 36),
```
Comment (msarahan) [inline conda_build/metadata.py]: We have made an explicit decision to not extend this going forward.  This PR will not be accepted with this change.

File: conda_build/metadata.py [hunk 2]
```diff
@@ -95,7 +95,7 @@ def ns_cfg(config):
     # there are times when python comes in as a tuple
     if not hasattr(py, 'split'):
         py = py[0]
-    py = int("".join(py.split('.')[:2]))
+    py = int("".join(py.split(' ')[0].split('.')[:2]))
```
Comment (msarahan) [inline conda_build/metadata.py]: Please add a comment here for values that you expect to see with the space.

File: conda_build/metadata.py (hunk 1 above)
Comment (isuruf) [reply conda_build/metadata.py]: Removed.

**Comment by isuruf** [comment]
For the noarch change, I have no idea how to add a test. It would require a python package with a `run_exports` section. Any ideas?

**Comment by msarahan** [comment]
You'll probably need to make a simple recipe.  You can build it as part of the test.  There's a few examples of that, especially around run_exports.  They make the tests slower, but it seems like the most stable, reliable way for tests to stay relevant over time (as opposed to building a package now and putting it somewhere).

**Comment by msarahan** [comment]
LGTM.  Thanks for the revisions.

**Comment by isuruf** [comment]
One test fails that might be related,
https://dev.azure.com/anaconda-conda/ccafa57b-d4b4-4525-b7e5-575388d7c030/_apis/build/builds/235/logs/115
with,
```
2020-02-06T20:28:07.5154326Z Traceback (most recent call last):
2020-02-06T20:28:07.5154675Z   File "/tmp/cb/popen-gw0/test_noarch_python_with_tests0/noarch_python_test_package_1581019389295/test_tmp/run_test.py", line 17, in <module>
2020-02-06T20:28:07.5154737Z     assert os.path.isdir(site_packages)
2020-02-06T20:28:07.5154780Z AssertionError
```

@msarahan, have you seen this before? or is this a new failure?

**Comment by mingwandroid** [comment]
Do we not really want to use a different run_export type here? Dropping all but the package name?

**Comment by isuruf** [comment]
@mingwandroid, I don't understand. Are you suggesting a different run_exports scheme ('noarch' in addition to 'weak', 'strong') that would be applied if the package is a dependency of a `noarch` package?

**Comment by mingwandroid** [comment]
I am.

**Comment by isuruf** [comment]
@mingwandroid, done.

**Comment by msarahan** [comment]
@mingwandroid I'm not sure where you were going with this, but please go ahead with your thoughts.  I'm fine with this PR now with or without the new run_exports type.

**Comment by mingwandroid** [comment]
Great. This is exactly what I meant.

**Comment by isuruf** [comment]
Tests pass now (except for unrelated test failures)

**Comment by mingwandroid** [comment]
Does anyone have any clues about https://dev.azure.com/anaconda-conda/conda-build/_build/results?buildId=239&view=logs&j=134db9d0-24f8-5cb4-e659-1d272ab2ea38&t=1bd5db4d-dc45-5f16-3e1b-4d645fd2ab7c&l=1548

I don't think it's related but I don't see any important differences there myself.

**Comment by marcelotrevisani** [comment]
> Does anyone have any clues about https://dev.azure.com/anaconda-conda/conda-build/_build/results?buildId=239&view=logs&j=134db9d0-24f8-5cb4-e659-1d272ab2ea38&t=1bd5db4d-dc45-5f16-3e1b-4d645fd2ab7c&l=1548
> 
> I don't think it's related but I don't see any important differences there myself.

It seems that one of the packages which the CRAN tests are using was archived. And because of that, you need more options to download it
We might mock those access to the CRAN repo with the current data which they return. After all, those tests are not testing the connection with the cran but the parser itself.

**Comment by isuruf** [comment]
Yes, those 2 tests are failing on master too and seems to be an issue with r-rsed being archived.

**Comment by isuruf** [comment]
ping

**Comment by isuruf** [comment]
Thanks @marcelotrevisani. Is there a plan for a release?

**Comment by marcelotrevisani** [comment]
> Thanks @marcelotrevisani. Is there a plan for a release?

Not that I am aware of... :/
Mike usually generated the releases, I can see with him how to generate it then

**Pull request merged**